### DISCUSSION TITLE
error UNHANDLED REJECTION spawn ENOMEM

### DISCUSSION THREAD

**Discussion by ppatel31** [gatsbyjs/gatsby#36897] [Help]
I see this error very offer but not able to find solution. i already increased memory using command NODE_OPTIONS="--max-old-space-size=12288"

#########
error UNHANDLED REJECTION spawn ENOMEM

 Error: spawn ENOMEM
 
- child_process:413 ChildProcess.spawn
node:internal/child_process:413:11
- node:child_process:700 spawn
node:child_process:700:9
 
- node:child_process:166 fork
node:child_process:166:10

- index.js:95 WorkerPool.startAll
[src]/[gatsby-worker]/dist/index.js:95:46

 - index.js:215 WorkerPool.restart
[src]/[gatsby-worker]/dist/index.js:215:10

- task_queues:96 processTicksAndRejections
node:internal/process/task_queues:96:5

not finished Writing page-data.json files to public directory - 0.934s
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by DSchau** [comment]
Hello @ppatel31!

Very hard to debug this without a simplified reproduction. Could you share more detail as to the environment in which this code is operating?

There are more details as to [creating a simplified reproduction here](https://gatsby.dev/reproduction)

Another thing that would be helpful for us is if you can paste the output of `gatsby info --clipboard` so we can see what plugins you're using.

**Comment by ppatel31** [comment]
here is the list of npmPackages i used

    gatsby: ^4.24.8 => 4.24.8
    gatsby-plugin-canonical-urls: ^4.24.0 => 4.24.0
    gatsby-plugin-google-tagmanager: ^4.24.0 => 4.24.0
    gatsby-plugin-no-sourcemaps: ^4.24.0 => 4.24.0
    gatsby-plugin-preconnect: ^1.3.0 => 1.3.0
    gatsby-plugin-robots-txt: ^1.7.1 => 1.7.1
    gatsby-plugin-sitemap: ^5.24.0 => 5.24.0
    gatsby-plugin-verify-bing: ^1.1.35 => 1.1.35
    gatsby-source-filesystem: ^4.24.0 => 4.24.0
    gatsby-source-graphql: ^4.24.0 => 4.24.0
    gatsby-transformer-json: ^4.24.0 => 4.24.0

I use AWS codebuild with linux machine. 15GB RAM, 8 CPU

increase memory using command NODE_OPTIONS="--max-old-space-size=12288" before start build

Generating around 2000 pages using JSON files.
